UI/Panels/Device/MFModulePanel.cs

```diff
@@ -93,20 +88,53 @@ private void DisplayDetails(Board board)
                 buttonDocs.Click += (s, e) => { Process.Start(CreateRedirectTarget(board.Info.Community.Docs)); };
             else
                 buttonDocs.Enabled = false;
-            
+
             if (board.Info.Community?.Support != null)
                 buttonSupport.Click += (s, e) => { Process.Start(CreateRedirectTarget(board.Info.Community.Support)); };
             else
                 buttonSupport.Enabled = false;
 
 
-            
+
             var defaultDeviceConfigFile = board.GetDefaultDeviceConfigFilePath();
-            if (defaultDeviceConfigFile != null && File.Exists(defaultDeviceConfigFile))
+            var defaultDeviceConfigFileExists = defaultDeviceConfigFile != null && File.Exists(defaultDeviceConfigFile);
+            var specificDeviceConfigs = board.Info.DeviceConfigs?.Where(p => File.Exists(board.GetDeviceConfigFilePath(p.File)));
+            var atLeastOneSpecificConfigExists = specificDeviceConfigs.Count() > 0;
+
+            if (defaultDeviceConfigFileExists || specificDeviceConfigs.Count() > 0)
             {
+                
                 buttonUploadDefaultConfig.Click += (s, e) =>
                 {
-                    UploadDefaultConfigRequested?.Invoke(this, defaultDeviceConfigFile);
+                    if (!atLeastOneSpecificConfigExists)
+                    {
+                        UploadDefaultConfigRequested?.Invoke(this, defaultDeviceConfigFile);
+                    }
+
+                    else if (specificDeviceConfigs.Count() == 1)
+                    {
+                        var profile = specificDeviceConfigs.First();
+                        var filePath = board.GetDeviceConfigFilePath(profile.File);
+                        UploadDefaultConfigRequested?.Invoke(this, filePath);
+                    }   
+
+                    // since we have more options, we present a context menu
+                    else
+                    {
+                        var menu = new ContextMenuStrip();
+                        foreach (var profile in board.Info.DeviceConfigs)
```

Comment:
Why is this using `board.Info.DeviceConfigs` instead of `specificDeviceConfigs` which already filtered out configs that don't have paths?